File: candydistribution.java
import java.util.*;
import java.io.*;
import java.math.BigInteger;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

        int t = Integer.parseInt(in.readLine());
        while (t-- > 0) {
            String line[] = in.readLine().split(" ");
            BigInteger K = new BigInteger(line[0]);
            BigInteger C = new BigInteger(line[1]);
            BigInteger gcd = K.gcd(C);
            BigInteger upperBound = BigInteger.TEN.pow(9);
            // System.err.format("(%s, %s) = %s\n", K, C, gcd);

            if (C.compareTo(K.add(BigInteger.ONE)) > 0) {
                System.out.println("IMPOSSIBLE");

            } else if (C.equals(BigInteger.ONE) && (K.compareTo(upperBound) >= 0)) {
                System.out.println("IMPOSSIBLE");

            } else if (C.equals(BigInteger.ONE)) {
                System.out.println(K.add(BigInteger.ONE));

            } else if (gcd.equals(BigInteger.ONE)) {
                BigInteger numBags = C.modInverse(K);
                // System.err.format("%s^-1 (mod %s) = %s\n", C, K, numBags);
                if (numBags.compareTo(upperBound) > 0) {
                    System.out.println("IMPOSSIBLE");
                } else if (numBags.equals(BigInteger.ZERO)) {
                    System.out.println(1);
                } else {
                    System.out.println(numBags);
                }

            } else {
                System.out.println("IMPOSSIBLE");
            }
            // System.err.println();
        }
        in.close();
    }
}

File: candydistribution.py
from sys import stdin

def egcd(a, b):
    x0, x1, y0, y1 = 0, 1, 1, 0
    while a != 0:
        q, b, a = b // a, a, b % a
        y0, y1 = y1, y0 - q * y1
        x0, x1 = x1, x0 - q * x1
    return b, x0, y0


t = int(stdin.readline())
for _ in range(t):
    line = stdin.readline().split()
    K = int(line[0])
    C = int(line[1])
    upperbound = 10**9
    xs = egcd(C, K)
    # if C == K+1:
    #     print(1)
    if C == 1 and K >= upperbound:
        print("IMPOSSIBLE")
    elif C == 1:
        print(K+1)
    elif K == 1:
        print(1)
    elif xs[0] != 1:
        print("IMPOSSIBLE")
    else:
        numbags = xs[1]%K
        if numbags == 0:
            print("IMPOSSIBLE")
        else:
            print(numbags)


